pkgs/games/r2modman/default.nix

```diff
@@ -0,0 +1,133 @@
+{ lib
+, stdenv
+, yarn
+, fetchYarnDeps
+, fixup_yarn_lock
+, nodejs
+, electron
+, fetchFromGitHub
+, gitUpdater
+, makeWrapper
+, makeDesktopItem
+, copyDesktopItems
+}:
+
+stdenv.mkDerivation rec {
+  pname = "r2modman";
+  version = "3.1.41";
+
+  src = fetchFromGitHub {
+    owner = "ebkr";
+    repo = "r2modmanPlus";
+    rev = "v${version}";
+    hash = "sha256-nnmzIpV+Rn2zBJDBUehcUpvh3U4jqGLLGoIoxlZy6SA=";
+  };
+
+  offlineCache = fetchYarnDeps {
+    yarnLock = "${src}/yarn.lock";
+    hash = "sha256-CXitb/b2tvTfrkFrFv4KP4WdmMg+1sDtC/s2u5ezDfI=";
+  };
+
+  nativeBuildInputs = [
+    yarn
+    fixup_yarn_lock
+    nodejs
+    makeWrapper
+    copyDesktopItems
+  ];
+
+  configurePhase = ''
+    runHook preConfigure
+
+    # Workaround for webpack bug
+    # https://github.com/webpack/webpack/issues/14532
+    export NODE_OPTIONS="--openssl-legacy-provider"
+    export HOME=$(mktemp -d)
+    yarn config --offline set yarn-offline-mirror $offlineCache
+    fixup_yarn_lock yarn.lock
+    yarn install --offline --frozen-lockfile --ignore-platform --ignore-scripts --no-progress --non-interactive
+    patchShebangs node_modules/
+
+    runHook postConfigure
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    yarn --offline quasar build --mode electron --skip-pkg
+
+    # Remove large directories in node_modules that are not needed at runtime.
```

Comment:
It's unfortunate yarn doesn't have an equivalent for `npm prune` to remove dev deps... maybe we could remove node_modules and reinstall with `--prod`?

Reply:
yarn does pruning automatically (see the help for [yarn prune](https://classic.yarnpkg.com/lang/en/docs/cli/prune/).  If upstream specified which of its dependencies were only needed at build-time, then your suggestion would work (and is what I do in [the derivation for heroic](https://github.com/NixOS/nixpkgs/blob/b9f18832c1256ab6f86bdcdf7bd83ddb70af9372/pkgs/games/heroic/default.nix#L68)).

Reply:
They do have a `devDependencies` section in their package.json, is that what you mean by separating the dependencies?

Reply:
Yes it is.  I don't know how I missed that. :facepalm:

Reply:
@aidalgol Ah nice, it seems to remove most of the deps, though `quasar` still seems to be present, is this needed at runtime?

Reply:
I don't think so, so that's annoying.